unlock-app/src/hooks/useCheckoutCommunication.ts

```diff
@@ -123,31 +124,21 @@ export const useCheckoutCommunication = () => {
   const { provider } = useProvider(AppConfig)
   const [user, setUser] = useState<string | undefined>(undefined)
 
-  // @ts-expect-error - attach to window
-  window.px = provider
+  const pushOrEmit = (kind: CheckoutEvents, payload?: Payload) => {
+    if (!parent) {
+      setOutgoingBuffer([...outgoingBuffer, { kind, payload }])
+    } else {
+      parent.emit(kind, payload)
+    }
+  }
 
-  const parent = usePostmateParent({
-    setConfig: (config: PaywallConfig) => {
-      setPaywallConfig(config)
-      setOauthConfig(undefined)
-    },
-    authenticate: (config: any) => {
-      setOauthConfig({
-        clientId: config?.clientId,
-        responseType: config?.responseType ?? '',
-        state: config?.state ?? '',
-        redirectUri: config?.redirectUri ?? '',
-      })
-      setPaywallConfig(undefined)
-    },
-    resolveMethodCall,
-    resolveOnEvent,
-    resolveOnEnable,
-    handleMethodCallEvent: async ({ id, params, method }: MethodCall) => {
-      // @ts-expect-error but we know it's there
-      const px = window.px
```

Comment:
This was attached to window because otherwise it wouldn't be be available when you try to access provider. Is that resolved with your changes?

Reply:
Hum, it was a debugging thing, right? We don't actually need window.px

Reply:
Iirc, the issue was that initially, provider is empty and when you initiate function in a new post instance - it would keep the provider as undefined due to closure since no new instance is ever created again due to the limitation of the library with multiple instances listening to same window. 

So provider was attached to window.px and updated later on when provider is not undefined without causing issues due to closure.